python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -17,6 +17,30 @@
 from cudf.utils.string import format_bytes
 
 
+def get_buffer_owner(data) -> Optional[BufferOwner]:
+    """Get the owner of `data`, if any exist
+
+    Search through the stack of data owners in order to find an
+    owner BufferOwner (incl. subclasses).
+
+    Parameters
+    ----------
+    data
+        The data object
+
+    Return
+    ------
+    BufferOwner or None
+        The owner of `data` if found otherwise None.
+    """
+
+    if isinstance(data, BufferOwner):
+        return data
+    if hasattr(data, "owner"):
+        return get_buffer_owner(data.owner)
```

Comment:
The public/private interface of objects at the Column/Buffer layer of cudf Python is pretty rough. Personally I've been punting on sorting out such questions until pylibcudf is ready since we'll have to rearchitect a lot of the internals around pylibcudf at that time anyway.